src/pgo/trans/intermediate/TLAExpressionCodeGenVisitor.java

```diff
@@ -172,7 +172,31 @@ public Expression visit(PGoTLABool pGoTLABool) throws RuntimeException {
 
 	@Override
 	public Expression visit(PGoTLACase pGoTLACase) throws RuntimeException {
-		throw new TODO();
+		VariableName result = null;
+		if (pGoTLACase.getOther() != null) {
+			result = builder.varDecl("result", pGoTLACase.getOther().accept(this));
+		} else {
+			UID uid = pGoTLACase.getArms().get(0).getResult().getUID();
+			result = builder.varDecl("result", typeMap.get(uid).accept(new PGoTypeGoTypeConversionVisitor()));
+		}
+		LabelName matched = builder.newLabel("matched");
+
+		for (PGoTLACaseArm caseArm : pGoTLACase.getArms()) {
```

Comment:
Does `pGoTLACase.getArms()` return the `OTHER` arm?

```pluscal
CASE x -> y
  [] z -> p
  [] OTHER -> other
```

Reply:
It does not (the types wouldn't match). The `OTHER` arm is accessed only by `pGoTLACase.getOther()`.

Reply:
It looks like you're missing codegen for the other branch when it is present.

Reply:
The `OTHER` arm is set to the default value of the result variable when present in the first `if`. It's confusing but seemed better than adding a check after the `if`s (which would have either required support for `else if` or a temporary boolean). Now that I'm using `goto`s I could move it below the `if`s but before the label if that would be more clear. What are your thoughts?

Reply:
Got it! I missed that part in the above code. I prefer the translated code to have similar logical structure to the TLA+ code. Could you please put code generation for the `OTHER` arm at the end?

Reply:
Ah, good to know. I just fixed it!